app/javascript/utilities/viewport.js

```diff
@@ -7,18 +7,35 @@
  *
  * @param {object} element - The HTML element to check
  * @param {number} [offsetTop=0] - Part of the screen to ignore counting from the top
- *
+ * @param {boolean} [isPartiallyVisible=false] - A boolean to flip the check between partial or completely visible in the viewport
  * @returns {boolean} isInViewport - true if the element is visible in the viewport
  */
-export function isInViewport(element, offsetTop = 0) {
+export function isInViewport(
+  element,
+  offsetTop = 0,
+  isPartiallyVisible = false,
+) {
```

Comment:
(refactor/blocking) - the updates here seem to work great! Just a small change requested here to make the developer experience a bit easier now that we have three params - let's change it to expect them in an object that we destructure like so:

```suggestion
export function isInViewport({
  element,
  offsetTop = 0,
  acceptPartiallyVisible = false,
}) {
```

So when we call this function it's clearer what the params are, e.g. `isInViewport({element: myElement, acceptPartiallyVisible: true})`

Reply:
Yes, plus it gives us flexibility as well. And also a lot more clearer. 👍